superset/proxy/api.py

```diff
@@ -0,0 +1,190 @@
+
+    openapi_spec_tag = "Proxy"
+
+    def __init__(self):
+        """
+        This is the init function for the ProxyRestAPI class
+        """
+        super().__init__()
+
+        self.ALFRED_SCOPE = os.environ.get("ALFRED_SCOPE")
+
+        self.ALFRED_URL = os.environ.get("ALFRED_URL")
+
+        if os.environ.get("FLASK_ENV") == "development":
+            self.SSL_CERT = os.environ.get("REQUESTS_CA_BUNDLE_DEV")
+        else:
+            self.SSL_CERT = os.environ.get("REQUESTS_CA_BUNDLE")
+
+    def attach_url(
+        self, response_code: int, app_url: str, err: bool, payload
+    ) -> Response:
+        """
+        This is a function that will attach the app URL with the response that is
+        being sent to the front-end (this will allow us to avoid hard coded URL's in
+        both the back-end and the front-end)
+        """
+        prep_payload = {"data": payload, "url": app_url, "err": err}
+
+        return self.response(response_code, payload=prep_payload)
+
+    def error_obtaining_token(
+        self, token_name: str, raised_exception: Exception
+    ) -> Response:
+        """
+        This is a function that returns an http response based on a passed in application name
+        and exception that was caught when trying to get an OBO token for an application
+        """
+        logger.error(
+            "Error obtaining on-behalf-of %s token: %s", token_name, raised_exception
+        )
+        return self.attach_url(
+            400,
+            self.ALFRED_URL,
+            True,
+            f"Error obtaining on-behalf-of {token_name} token: {raised_exception}",
+        )
+
+    def error_obtaining_response(
+        self, token_name: str, raised_exception: Exception
+    ) -> Response:
+        """
+        This is a function that returns an http response based on a passed in application name
+        and exception that was caught when trying to get a response from an application
+        """
+        logger.error("Error obtaining %s response: %s", token_name, raised_exception)
+        return self.attach_url(
+            400,
+            self.ALFRED_URL,
+            True,
+            f"Error obtaining {token_name} response: {raised_exception}",
+        )
+
+    @expose("/alfred/user_id/<string:user_id>", methods=["GET"])
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **_kwargs: f"{self.__class__.__name__}.get",
+        log_to_statsd=False,  # pylint: disable-arguments-renamed
+    )
+    def get_userid(self, user_id: str, **_kwargs: Any) -> Response:
+        """
+        This is a function which will obtain an Alfred OBO token based on the
+        current logged in user, and will then send a request to Alfred to see
+        if the passed in user_id is in any reports/incidents
+        """
+        if (self.ALFRED_SCOPE is None or self.ALFRED_URL is None):
+            return self.error_obtaining_response("Alfred", "No Alfred Scope and/or No Alfred URL")
```

Comment:
Would it be worth breaking No Alfred Scope and No Alfred URL into their own code paths or will there be a stacktrace that makes it obvious which problem we're dealing with?

Reply:
I can definitely split them up, I only kept them together to reduce the number of reused code lines

Reply:
Fixed in the latest commit. Code was changed to be:

```suggestion
        if (self.ALFRED_SCOPE is None and self.ALFRED_URL is None):
            return self.error_obtaining_response("Alfred", "No Alfred Scope and No Alfred URL")
        if (self.ALFRED_SCOPE is None):
            return self.error_obtaining_response("Alfred", "No Alfred Scope")
        if (self.ALFRED_URL is None):
            return self.error_obtaining_response("Alfred", "No Alfred URL")
```